### DISCUSSION TITLE
timer.setOverflow have some freeze and delay

### DISCUSSION THREAD

**Discussion by EeeLo** [stm32duino/Arduino_Core_STM32#2006] [Q&A]
I'm using Teensystep library to control steper motor, Arduino IDE and nucleo-f446re board,but there have some problem with my project.my code set new speed value frequently,I track the timer code is .setOverflow(), but sometimes the code already call .setOverflow() and motor freeze some millisecond and then to run,any good idea?
this is output data pic from setOverflow():
![image](https://user-images.githubusercontent.com/6959959/231933477-84620fe2-dc61-4081-8115-899437ea1c41.png)

we can see that the code already to call successful with new value,

this is Teensystep library:
```
void TimerField::setStepFrequency(unsigned f)
{
  
  if(f == 0){
    //stepTimerStop();
    return;
  }
  
  stepTimer.setOverflow(f, HERTZ_FORMAT);
  Serial1.println("Set overflow:" + String(f));
}
```

**Comment by fpistm** [comment]
Hi @EeeLo 
We don't really understand your point.
Digging to Teensystep library, we found you have opened an issue: https://github.com/luni64/TeensyStep/issues/152
and maintainer does not understand nor be able to build your code. 

It seems the STM32 is not officially supported and a dedicated dev branch for STM32 exists. As you stated you used and modified it.
Moreover, this branch seems not maintained since 2 years and probably not updated against latest version of the core.

Here you didn't provide useful information to be able to help. No code, log seems not useful.

As you used third party library with a non-maintained branch and modified by you, we can't help. If you are able to reproduce without this library we will please to help or provide to Teensystep maintainer enough information to be able to help on the issue you closed.

**Reply by EeeLo** [reply]
Sorry for my info, I'm using Teensystep to control steper motor as you know,the Teensystep have a mode named "[RotateControl](https://luni64.github.io/TeensyStep/documentation/rotateControl/)" that's working for speed update with motor running.But there have some freeze with my motor when I update speed frequent sometimes,so I tracked the code and then find they call setOverflow function to update step timer.
And I add some code to print the value every code call Timer.setOverflow() function, the print seems very good when speed changed,so I think the problem is from somewhere between stm32duino library or hardware of by board(nucleo-f446).
Can you tell me what's happened when I call Timer.setOverflow frequent? Why does my motor running with some freeze sometimes? I don't have logical analysizer now.
the code using JY901 senor to get new angle and update speed to control motor compensate.
